x/bank/keeper/grpc_query.go

```diff
@@ -0,0 +1,72 @@
+package keeper
+
+import (
+	"context"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+var _ types.QueryServer = BaseKeeper{}
+
+// Balance implements the Query/Balance gRPC method
+func (q BaseKeeper) Balance(c context.Context, req *types.QueryBalanceRequest) (*types.QueryBalanceResponse, error) {
+	if req == nil {
+		return nil, sdkerrors.ErrInvalidRequest
```

Comment:
I believe status (https://pkg.go.dev/google.golang.org/grpc@v1.29.1/status?tab=doc) for errors here would be better?

Reply:
Great idea @marbar3778!

Reply:
They're updated now to use `status.Errorf`. How does that look?

Reply:
Looks good! Thanks